spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -40,36 +42,41 @@
 
 	private final BinderFactory<MessageChannel> binderFactory;
 
-	private final Properties producerProperties;
+	private final ChannelBindingServiceProperties channelBindingServiceProperties;
 
-	private DefaultListableBeanFactory beanFactory;
+	private ConfigurableListableBeanFactory beanFactory;
 
-	public BinderAwareChannelResolver(BinderFactory<MessageChannel> binderFactory, Properties producerProperties) {
+	public BinderAwareChannelResolver(BinderFactory<MessageChannel> binderFactory,
+			ChannelBindingServiceProperties channelBindingServiceProperties) {
+		Assert.notNull(binderFactory, "'binderFactory' cannot be null");
 		this.binderFactory = binderFactory;
-		this.producerProperties = producerProperties;
+		this.channelBindingServiceProperties = channelBindingServiceProperties;
 	}
 
 	@Override
 	public void setBeanFactory(BeanFactory beanFactory) {
 		super.setBeanFactory(beanFactory);
-		if (beanFactory instanceof ConfigurableBeanFactory) {
-			this.beanFactory = (DefaultListableBeanFactory) beanFactory;
+		if (beanFactory instanceof ConfigurableListableBeanFactory) {
+			this.beanFactory = (ConfigurableListableBeanFactory) beanFactory;
 		}
 	}
 
 	@Override
 	public MessageChannel resolveDestination(String name) {
 		MessageChannel channel = null;
+		DestinationResolutionException destinationResolutionException;
 		try {
 			return super.resolveDestination(name);
 		}
 		catch (DestinationResolutionException e) {
+			destinationResolutionException = e;
 		}
 		synchronized (this) {
 			try {
```

Comment:
Seems like the superclass call is duplicate, once inside the synchronized block, and once outside. Is this a double-check locking?